src/components/Calculator.js

```diff
@@ -1,48 +1,63 @@
-import React from 'react';
+import React, { useState } from 'react';
+import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
+const Calculator = () => {
+  const [calc, setcalc] = useState({
+    next: '',
+    total: 0,
+    operation: '',
+  });
+  const { next, total, operation } = calc;
+  const handleClick = (e) => {
+    e.preventDefault();
+    setcalc((state) => calculate({
+      next: state.next,
+      total: state.total,
+      operation: state.operation,
+    },
+    e.target.innerHTML));
+  };
 
-  render() {
-    return (
-      <div className="main-wrap">
-        <div className="calc-wrap">
-          <input type="number" id="input" placeholder="0" />
-          <div id="row1">
-            <button type="button" className="op">Ac</button>
-            <button type="button" className="op">+/-</button>
-            <button type="button" className="op">%</button>
-            <button type="button" className="operator">÷</button>
-          </div>
-          <div id="row2">
-            <button type="button" className="number" id="7">7</button>
-            <button type="button" className="number" id="8">8</button>
-            <button type="button" className="number" id="9">9</button>
-            <button type="button" className="operator">*</button>
-          </div>
-          <div id="row3">
-            <button type="button" className="number" id="4">4</button>
-            <button type="button" className="number" id="5">5</button>
-            <button type="button" className="number" id="6">6</button>
-            <button type="button" className="operator">-</button>
-          </div>
+  return (
+    <div className="calc-wrap">
+      <div id="input">
+        {total}
+        {' '}
+        {operation}
+        {' '}
+        {next}
+      </div>
+      <div id="row1">
+        <button type="button" className="op" onClick={handleClick}>AC</button>
+        <button type="button" className="op" onClick={handleClick}>+/-</button>
+        <button type="button" className="op" onClick={handleClick}>%</button>
+        <button type="button" className="operator" onClick={handleClick}>÷</button>
+      </div>
+      <div id="row2">
+        <button type="button" className="number" id="7" onClick={handleClick}>7</button>
+        <button type="button" className="number" id="8" onClick={handleClick}>8</button>
+        <button type="button" className="number" id="9" onClick={handleClick}>9</button>
+        <button type="button" className="operator" onClick={handleClick}>*</button>
```

Comment:
- [ ] Please the multiplication operator(`*`) has an issue, the app crashes each time I try to multiply two numbers.  Please checkout the console messages in the screenshot below and fix the issue.

![multiplication_operator](https://user-images.githubusercontent.com/65852621/164256104-135f7af6-8441-431c-8e2e-0b1d7691b865.png)